opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfigurationTest.java

```diff
@@ -62,26 +82,35 @@ public MockTracer tracer() {
     @Test
     public void testRequestIsTraced() {
         testRestTemplate.getForEntity("/hello", String.class);
-        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(2));
-
-        assertThat(mockTracer.finishedSpans()).hasSize(2);
+        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
+        assertThat(mockTracer.finishedSpans()).hasSize(1);
     }
 
     @Test
-    public void testOnlyUrlPatternsIsTraced() {
+    public void testOnlyUrlPatternsIsTraced() throws InterruptedException {
         testRestTemplate.getForEntity("/skip", String.class);
-        Awaitility.await().pollDelay(500, TimeUnit.MILLISECONDS).until(reportedSpansSize(), IsEqual.equalTo(0));
+        skipCountDownLatch.await();
+
+        assertThat(mockTracer.finishedSpans()).hasSize(0);
+        assertThat(Mockito.mockingDetails(mockDecorator1).getInvocations()).hasSize(0);
+    }
+
+    @Test
+    public void testExcluded() throws InterruptedException {
+        testRestTemplate.getForEntity("/excluded", String.class);
+        excludeCountDownLatch.await();
 
         assertThat(mockTracer.finishedSpans()).hasSize(0);
         assertThat(Mockito.mockingDetails(mockDecorator1).getInvocations()).hasSize(0);
+        assertThat(Mockito.mockingDetails(mockDecorator2).getInvocations()).hasSize(0);
     }
 
     @Test
     public void testDecoratedProviderIsUsed() {
         testRestTemplate.getForEntity("/hello", String.class);
-        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(2));
+        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
 
-        assertThat(mockTracer.finishedSpans()).hasSize(2);
+        assertThat(mockTracer.finishedSpans()).hasSize(1);
         assertThat(Mockito.mockingDetails(mockDecorator1).getInvocations()).hasSize(2);
```

Comment:
Is this dependent on order tests invoked? Should the mocks be reset just in case.

Reply:
it's probably not because it works.

Reply:
might still be best to reset them, to avoid any unexpected issues later, but I'll leave that up to you.

Reply:
Docs says that reset is applied on `MockBeans` automatically after the test.